aggsender/aggsender.go

```diff
@@ -526,26 +522,30 @@ func (a *AggSender) pollValidators(
 			continue
 		}
 
-		signatures = append(signatures, res.signature)
-		if uint32(len(signatures)) >= signaturesThreshold {
+		multisig.Signatures = append(multisig.Signatures, agglayertypes.ECDSAMultisigEntry{
+			Index:     res.validator.Index(),
+			Signature: res.signature,
+		})
+
+		if uint32(len(multisig.Signatures)) >= signaturesThreshold {
 			cancel() // signal other goroutines to stop early
 		}
 	}
 
-	if uint32(len(signatures)) < signaturesThreshold {
+	if uint32(len(multisig.Signatures)) < signaturesThreshold {
 		metrics.MultiSigThresholdNotReached()
 
 		if len(errs) > 0 {
-			return signatures, errors.Join(errs...)
+			return multisig, errors.Join(errs...)
```

Comment:
I'm having some second thoughts here, should we return a nil for the multisig?
```suggestion
			return nil, errors.Join(errs...)
```

Reply:
Fix: https://github.com/agglayer/aggkit/pull/926/commits/8ba71cbfe84adf2ef596d7080f2893b4c9a1a564